processing/src/main/java/org/apache/druid/math/expr/ApplyFunction.java

```diff
@@ -476,6 +476,44 @@ private <T> Stream<T> filter(T[] array, LambdaExpr expr, SettableLambdaBinding b
     }
   }
 
+  /**
+   * Extended version of {@link FilterFunction} to return a null expr if filtered result turns out to be empty
+   */
+  class FilterOnlyFunction extends FilterFunction
+  {
+    static final String NAME = "filter_only";
+
+    @Override
+    public String name()
+    {
+      return NAME;
+    }
+
+    @Override
+    public ExprEval apply(LambdaExpr lambdaExpr, List<Expr> argsExpr, Expr.ObjectBinding bindings)
+    {
+      Expr arrayExpr = argsExpr.get(0);
+      ExprEval arrayEval = arrayExpr.eval(bindings);
+
+      Object[] array = arrayEval.asArray();
+      if (array == null) {
+        return ExprEval.of(null);
```

Comment:
nit: this should probably return `ExprEval.ofArray(arrayEval.asArrayType(), null)` so it doesn't change the type to STRING. This is a bug in `FilterFunction` too..